src/register/data/optimizelyExperiment/useMultiStepRegistrationExperimentVariation.jsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+
+import {
+  activateMultiStepRegistrationExperiment,
+  getMultiStepRegistrationExperimentVariation,
+  NOT_INITIALIZED,
+} from './helper';
+import { COMPLETE_STATE } from '../../../data/constants';
+
+/**
+ * This hook returns activates multi step registration experiment and returns the experiment
+ * variation for the user.
+ */
+const useMultiStepRegistrationExperimentVariation = (
+  initExpVariation,
+  registrationEmbedded,
+  tpaHint,
+  currentProvider,
+  thirdPartyAuthApiStatus,
+) => {
+  const [variation, setVariation] = useState(initExpVariation);
+
+  useEffect(() => {
+    if (initExpVariation || registrationEmbedded || !!tpaHint || !!currentProvider
+        || thirdPartyAuthApiStatus !== COMPLETE_STATE) {
+      return variation;
+    }
+
+    const getVariation = () => {
+      const expVariation = getMultiStepRegistrationExperimentVariation();
+      if (expVariation) {
+        setVariation(expVariation);
+      } else {
+        // This is to handle the case when user dont get variation for some reason, the register page
+        // shows unlimited spinner.
+        setVariation(NOT_INITIALIZED);
+      }
+    };
+
+    activateMultiStepRegistrationExperiment();
+
+    const timer = setTimeout(getVariation, 300);
```

Comment:
Would it be enough time to activate the experiment or should we catch the fall back on 500 ms if the experiment is not activated within 300 ms?